apps/api/src/app/shared/services/in-memory-provider/in-memory-provider.service.ts

```diff
@@ -41,13 +42,31 @@ export class InMemoryProviderService {
     return this.isClusterMode() ? this.inMemoryClusterProviderSetup() : this.inMemoryProviderSetup();
   }
 
-  public isClientReady(): boolean {
-    if (!this.inMemoryProviderClient) {
-      return false;
+  public async delayUntilReadiness(): Promise<void> {
+    let times = 0;
+    const retries = process.env.IN_MEMORY_PROVIDER_SERVICE_READINESS_TIMEOUT_RETRIES
+      ? Number(process.env.IN_MEMORY_PROVIDER_SERVICE_READINESS_TIMEOUT_RETRIES)
+      : 10;
+    const timeout = process.env.IN_MEMORY_PROVIDER_SERVICE_READINESS_TIMEOUT
+      ? Number(process.env.IN_MEMORY_PROVIDER_SERVICE_READINESS_TIMEOUT)
+      : 100;
```

Comment:
Giving flexibility in case in the future we find ourselves in cold starts in a certain environment and we want to configure the waiting time.